docs/content/TextArea.mdx

```diff
@@ -0,0 +1,187 @@
+---
+componentId: textarea
+title: TextArea
+description: Use TextArea for multi-line text input form fields
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/TextArea.tsx
+storybook: /react/storybook?path=/story/forms-textarea--default
+---
+
+import {TextArea} from '@primer/react'
+
+```js
+import {TextArea} from '@primer/react'
+```
+
+## Examples
+
+<Note variant="warning">
+
+TextArea components **must always** be accompanied by a corresponding label to improve support for assistive
+technologies. Examples below are provided for conciseness and may not reflect accessibility best practices.
+
+</Note>
+
+### Basic
+
+```javascript live noinline
+const TextAreaExample = () => {
+  // Running in controlled mode (recommended)
+  const [value, setValue] = React.useState('')
+
+  const handleChange = event => {
+    setValue(event.target.value)
+  }
+
+  return <TextArea onChange={handleChange}>{value}</TextArea>
+}
+
+render(<TextAreaExample />)
+```
+
+### Uncontrolled mode
+
+```javascript live noinline
+const TextAreaExample = () => {
+  const ref = React.useRef()
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    if (!ref.current.value) {
+      alert(`Enter a value into the TextArea and press submit`)
+      return
+    }
+
+    alert(`Current TextArea value: ${ref.current.value}`)
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <TextArea ref={ref} defaultValue="Set the initial state in uncontrolled mode using the defaultValue prop" />
+      <br />
+      <Button type="submit">Submit</Button>
+    </form>
+  )
+}
+
+render(<TextAreaExample />)
+```
+
+### Displaying form validation state
+
+```jsx live
+<>
+  <Box as="label" for="success-state" sx={{display: 'block'}}>
+    Success state:
+  </Box>
+  <TextArea id="success-state" validationStatus="success" />
+  <Box as="label" for="error-state" sx={{display: 'block', mt: 5}}>
+    Error state:
+  </Box>
+  <TextArea id="error-state" validationStatus="error" />
+</>
+```
+
+### Inactive
```

Comment:
Can we call this disabled?

Reply:
Can do. It's for more inclusive language. This came up on product-design slack around accessible language, for example: https://polaris.shopify.com/content/accessible-and-inclusive-language#enable-disable. I think it's a good idea here, but the underlying `disabled` attribute hasn't changed, just the contextual semantics.